hq/app/schedule/IamAudit.scala

```diff
@@ -1,30 +1,56 @@
 package schedule
 
-import com.gu.anghammarad.models.{Target, AwsAccount => Account}
+import com.gu.anghammarad.models.{Notification, Target, AwsAccount => Account}
 import config.Config.{iamAlertCadence, iamHumanUserRotationCadence, iamMachineUserRotationCadence}
 import logic.DateUtils
 import model._
 import org.joda.time.{DateTime, Days}
 import play.api.Logging
 import schedule.IamMessages._
-import schedule.IamNotifier.createNotification
+import schedule.IamNotifier.notification
 import utils.attempt.FailedAttempt
 
 object IamAudit extends Logging {
 
-  def makeIamNotification(flaggedCredsToNotify: Map[AwsAccount, Seq[IAMAlertTargetGroup]]): List[IamNotification] = {
+  def makeNotification(flaggedCredsToNotify: Map[AwsAccount, Seq[IAMAlertTargetGroup]]): List[IamNotification] = {
     flaggedCredsToNotify.toList.flatMap { case (awsAccount, targetGroups) =>
       if (targetGroups.isEmpty) {
         logger.info(s"found no IAM user issues for ${awsAccount.name}. No notification required.")
         None
       } else {
-        targetGroups.map { tg =>
-          logger.info(s"for ${awsAccount.name}, generating iam notification message for ${tg.users.length} user(s) with outdated keys and and/or missing mfa")
-          val (usersToReceiveWarningAlerts, usersToReceiveFinalAlerts) = sortUsersIntoWarningOrFinalAlerts(tg.users)
-          createNotifications(warning = true, usersToReceiveWarningAlerts, awsAccount, tg.targets) ++ createNotifications(warning = false, usersToReceiveFinalAlerts, awsAccount, tg.targets)
+        targetGroups.map{ tg =>
+          val (warningNotification, finalNotification) = createWarningAndFinalNotification(tg, awsAccount)
+          val users = createIamAuditUsers(tg.users, awsAccount)
+          IamNotification(warningNotification, finalNotification, users)
         }
       }
-    }.flatten
+    }
+  }
+
+  def createIamAuditUsers(users: Seq[VulnerableUser], account: AwsAccount): Seq[IamAuditUser] = {
+    users.map { user =>
+      IamAuditUser(
+        Dynamo.createId(account, user.username), account.name, user.username,
+        List(IamAuditAlert(DateTime.now, createDeadlineIfMissing(user.disableDeadline)))
+      )
+    }
+  }
+
+  def createWarningAndFinalNotification(tg: IAMAlertTargetGroup, awsAccount: AwsAccount): (Option[Notification], Option[Notification]) = {
+    logger.info(s"for ${awsAccount.name}, generating iam notification message for ${tg.users.length} user(s) with outdated keys and and/or missing mfa")
+    val (usersToReceiveWarningAlerts, usersToReceiveFinalAlerts) = sortUsersIntoWarningOrFinalAlerts(tg.users)
+
+    val warningNotifications =
+      if (usersToReceiveWarningAlerts.nonEmpty)
```

Comment:
you can use map here as well I think

Reply:
Hey, we discussed putting this if statement logic into the `createNotification` function, which is a great idea except warning and final notifications need different boolean parameters to differentiate the type of notification created. I think this will have to stay as is for now.